config/validator.go

```diff
@@ -142,3 +142,41 @@ func RequestHeaderContentTypeValidator(req http.Request, requiredContentType str
 	}
 	return 0
 }
+
+
+// DomainsEntryLogicalValidator checks the correctness of "Domains" part of configuration file
+func (c *YamlConfig) DomainsEntryLogicalValidator() (valid bool, validationErrors map[string][]error) {
+	errList := make([]error, 0)
+	var domains []string
+	for _, regionConf := range c.Regions {
+		for _, regionDomain := range regionConf.Domains {
+			for _, alreadyDefinedDomain := range domains {
```

Comment:
In case previous comment on that line has missed:
You may move body of that for block to separate function for clarity (one indentation less)
eg func isAlreadyInDefinedDomains(current string, verifiedDomains []string)
(and maybe rename var domain to verifiedDomains)